src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -2179,6 +2179,25 @@ private void EmitThisForMethodCall(Expression node)
             CompileAddress(node, -1);
         }
 
+        private static bool ShouldWritebackNode(Expression node)
+        {
+            if (!node.Type.GetTypeInfo().IsValueType)
+                return false;
+            switch (node.NodeType)
+            {
+                case ExpressionType.Parameter:
+                case ExpressionType.Call:
+                case ExpressionType.ArrayIndex:
```

Comment:
An a related note, the subsequent `case` for `Unbox` is an exact duplicate of the immediately following `default`. I'd consider removing it to be clearer, but maybe other people find the explicit case clearer.